src/rdb/rdb_raft.c

```diff
@@ -2521,6 +2521,18 @@ rdb_raft_resign(struct rdb *db, uint64_t term)
 	D_ASSERTF(rc == 0, ""DF_RC"\n", DP_RC(rc));
 }
 
+/* TODO: update raft.h in daos-stack/raft repo */
+extern int raft_election_start(raft_server_t *me_);
+
+/* Call new election (campaign to be leader) */
+int
+rdb_raft_campaign(struct rdb *db)
+{
+	D_DEBUG(DB_MD, DF_DB": calling election from current term %d\n",
+		DP_DB(db), raft_get_current_term(db->d_raft));
+	return raft_election_start(db->d_raft);
```

Comment:
As you suggested in a mail, we need to first check

    if (i_am_not_follower)
        return -DER_NOTFOLLOWER;

Additionally, we might want to call rdb_raft_save_state and rdb_raft_check_state before and after calling raft_election_start, for consistency reasons (and perhaps error handling?).

Reply:
Latest version of patch checks for not follower. As discussed in email it returns 0 if the caller is not a follower. This is similar to rdb_resign() that checks for leader status and turns that into a no-op if the condition is not met.

Reply:
OK.